lib/cli-engine.js

```diff
@@ -154,105 +193,62 @@ function calculateStatsPerRun(results) {
+ * @param {string} text The source code to verify.
+ * @param {string} filePath The path to the file of `text`.
+ * @param {ConfigArray} config The config.
+ * @param {boolean} fix If `true` then it does fix.
+ * @param {boolean} allowInlineConfig If `true` then it uses directive comments.
+ * @param {boolean} reportUnusedDisableDirectives If `true` then it reports unused `eslint-disable` comments.
+ * @param {Linter} linter The linter instance to verify.
+ * @returns {LintResult} The result of linting.
  * @private
  */
-function processText(text, configHelper, filename, fix, allowInlineConfig, reportUnusedDisableDirectives, linter) {
-    let filePath,
-        fileExtension,
-        processor;
-
-    if (filename) {
-        filePath = path.resolve(filename);
-        fileExtension = path.extname(filename);
-    }
-
-    const effectiveFilename = filename || "<text>";
-
-    debug(`Linting ${effectiveFilename}`);
-    const config = configHelper.getConfig(filePath);
-
-    if (config.plugins) {
-        configHelper.plugins.loadAll(config.plugins);
-    }
-
-    if (config.parser) {
-        if (!path.isAbsolute(config.parser)) {
-            throw new Error(`Expected parser to be an absolute path but found ${config.parser}. This is a bug.`);
-        }
-        linter.defineParser(config.parser, require(config.parser));
-    }
-
-    const loadedPlugins = configHelper.plugins.getAll();
-
-    for (const plugin in loadedPlugins) {
-        if (loadedPlugins[plugin].processors && Object.keys(loadedPlugins[plugin].processors).indexOf(fileExtension) >= 0) {
-            processor = loadedPlugins[plugin].processors[fileExtension];
-            break;
+function verifyText(
+    text,
+    filePath,
+    config,
+    fix,
+    allowInlineConfig,
+    reportUnusedDisableDirectives,
+    linter
+) {
+    debug(`Lint ${filePath}`);
+
+    // Verify.
+    const { fixed, messages, output } = linter.verifyAndFix(
+        text,
+        config,
+        {
+            allowInlineConfig,
+            filename: filePath,
+            fix,
+            reportUnusedDisableDirectives
         }
-    }
+    );
 
-    const autofixingEnabled = typeof fix !== "undefined" && (!processor || processor.supportsAutofix);
-    const fixedResult = linter.verifyAndFix(text, config, {
-        filename: effectiveFilename,
-        allowInlineConfig,
-        reportUnusedDisableDirectives,
-        fix: !!autofixingEnabled && fix,
-        preprocess: processor && (rawText => processor.preprocess(rawText, effectiveFilename)),
-        postprocess: processor && (problemLists => processor.postprocess(problemLists, effectiveFilename))
-    });
-    const stats = calculateStatsPerFile(fixedResult.messages);
+    const basename = path.basename(filePath, path.extname(filePath));
+    const resultFilePath = basename.startsWith("<") && basename.endsWith(">")
```

Comment:
Yes.

`ConfigArray#extractConfig(filePath)` method requires an absolute path, but `Linter` object doesn't know CWD. So CLIEngine passes an absolute path always. If a user didn't specify the file path on `CLIEngine`'s methods, `filePath` is an absolute path like `/path/of/cwd/<text>`. On the other hand, the lint result uses only `<text>` part.

Reply:
Could we explicitly check for `<text>` instead (e.g. when converting to an absolute path)? It seems like this could be a confusing edge case if someone has files that start with `<` and end with `>` for some reason.